src/components/NcKbd/NcKbd.vue

```diff
@@ -0,0 +1,243 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+import { computed } from 'vue'
+import { t } from '../../l10n.ts'
+import { isMacos } from '../../utils/platform.ts'
+
+const {
+	symbol = undefined,
+	mac = isMacos,
+} = defineProps<{
+	/**
+	 * Key name or symbol to display.
+	 * For common special keys (CTRL, ALT, SHIFT, Arrow keys, etc.) it will display the symbol on macOS and the localized name on Windows/Linux.
+	 */
+	symbol?: 'ArrowUp' | 'ArrowDown' | 'ArrowLeft' | 'ArrowRight' | 'Control' | 'Alt' | 'Shift' | 'Enter' | 'Space' | 'Tab' | 'Delete' | 'Escape' | (string & {})
+	/**
+	 * Explicitly use macOS (true) or Windows/Linux (false) key symbols.
+	 * By default it uses the OS detected from the user agent.
+	 */
+	mac?: boolean | undefined
+}>()
+
+defineSlots<{
+	default?: Slot
+}>()
+
+/**
+ * Map of special key names to symbols or localized names:
+ * - macOS uses symbols instead of names
+ * - Windows/Linux uses localized names
+ * - In ternary expressions, // TRANSLATORS comments only works for the second operand, but not for `else`
+ */
+const labels = computed(() => ({
+	ArrowUp: '↑',
+	ArrowDown: '↓',
+	ArrowLeft: '←',
+	ArrowRight: '→',
+	Control: !mac
+		? t('Ctrl') // TRANSLATORS: Ctrl key on keyboard (Windows/Linux)
+		: '⌘',
+	Alt: !mac
+		? t('Alt') // TRANSLATORS: Alt key on keyboard (Windows/Linux)
+		: '⌥',
+	Shift: !mac
+		? t('Shift') // TRANSLATORS: Shift key on keyboard
+		: '⇧',
+	Enter: !mac
+		? t('Enter') // TRANSLATORS: Enter key on keyboard
+		: '⏎',
+	Tab: !mac
+		? t('Tab') // TRANSLATORS: Tab key on keyboard
+		: '⇥',
+	Delete: !mac
+		? t('Delete') // TRANSLATORS: Delete key on keyboard
+		: '⌫',
+	Escape: !mac
+		? t('Escape') // TRANSLATORS: Escape key on keyboard
+		: '⎋',
+	Space: t('Space'), // TRANSLATORS: Space key on keyboard
+} as const))
+
+const label = computed(() => (symbol && labels.value[symbol]) || symbol)
+</script>
+
+<template>
+	<kbd :class="$style.kbd">
+		<slot>
+			{{ label }}
+		</slot>
+	</kbd>
+</template>
+
+<style lang="scss" module>
+.kbd {
+	display: inline-flex;
+	align-items: center;
+	justify-content: center;
+	height: var(--default-clickable-area);
```

Comment:
IMHO it looks weird that those "keys" have different width.
You can see it in your example on the `<-` and the `^` keys.
Maybe we should - for those single letter keys - add a min-width here, like:

```suggestion
	height: var(--default-clickable-area);
	min-width: var(--default-clickable-area);
```

Reply:
Oh right, min-width is a thing. Yeah, that would totally solve the issue here ^^

Reply:
Good point, done.